vm/actor/tests/common/mod.rs

```diff
@@ -0,0 +1,505 @@
+        return Ok(resolved.clone());
+    }
+
+    fn get_actor_code_cid(&self, addr: &Address) -> Result<Cid, ActorError> {
+        self.require_in_call();
+        let ret = self.actor_code_cids.get(&addr).unwrap();
+        Ok(ret.clone())
+    }
+
+    fn get_randomness(
+        _personalization: DomainSeparationTag,
+        _rand_epoch: ChainEpoch,
+        _entropy: &[u8],
+    ) -> Randomness {
+        unimplemented!()
+    }
+
+    fn create<C: Cbor>(&mut self, obj: &C) -> Result<(), ActorError> {
+        if self.state.is_some() == true {
+            return Err(self.abort(
+                ExitCode::SysErrorIllegalActor,
+                "state already constructed".to_owned(),
+            ));
+        }
+        self.state = Some(self.store.put(obj, Blake2b256).unwrap());
+        Ok(())
+    }
+
+    fn state<C: Cbor>(&self) -> Result<C, ActorError> {
+        Ok(self
+            .store
+            .get(&self.state.as_ref().unwrap())
+            .unwrap()
+            .unwrap())
+    }
+
+    fn transaction<C: Cbor, R, F>(&mut self, f: F) -> Result<R, ActorError>
+    where
+        F: FnOnce(&mut C, &Self) -> R,
+    {
+        if self.in_transaction {
+            return Err(self.abort(ExitCode::SysErrorIllegalActor, "nested transaction"));
+        }
+        let mut read_only = self.state()?;
+        self.in_transaction = true;
+        let ret = f(&mut read_only, &self);
+        self.state = Some(self.put(&read_only).unwrap());
+        Ok(ret)
+    }
+
+    fn store(&self) -> &BS {
+        self.store
+    }
+
+    fn send(
+        &mut self,
+        to: &Address,
+        method: MethodNum,
+        params: &Serialized,
+        value: &TokenAmount,
+    ) -> Result<Serialized, ActorError> {
+        self.require_in_call();
+        if self.in_transaction {
+            return Err(self.abort(
+                ExitCode::SysErrorIllegalActor,
+                "side-effect within transaction",
+            ));
+        }
+
+        assert!(
+            !self.expect_sends.is_empty(),
+            "unexpected expectedMessage to: {:?} method: {:?}, value: {:?}, params: {:?}",
+            to,
+            method,
+            value,
+            params
+        );
+
+        let expected_msg = self.expect_sends[0].clone();
+
+        assert!(&expected_msg.to == to && expected_msg.method == method && &expected_msg.params == params && &expected_msg.value == value, "expectedMessage being sent does not match expectation.\nMessage -\t to: {:?} method: {:?} value: {:?} params: {:?}\nExpected -\t {:?}", to, method, value, params, &self.expect_sends[0]);
+
+        if value > &self.balance {
+            return Err(self.abort(
+                ExitCode::SysErrSenderStateInvalid,
+                format!(
+                    "cannot send value: {:?} exceeds balance: {:?}",
+                    value, self.balance
+                ),
+            ));
+        }
+        let _ = self.expect_sends.pop_front();
```

Comment:
```suggestion
        let expected_msg = self.expect_sends.pop_front().unwrap();

        assert!(&expected_msg.to == to && expected_msg.method == method && &expected_msg.params == params && &expected_msg.value == value, "expectedMessage being sent does not match expectation.\nMessage -\t to: {:?} method: {:?} value: {:?} params: {:?}\nExpected -\t {:?}", to, method, value, params, &self.expect_sends[0]);

        if value > &self.balance {
            return Err(self.abort(
                ExitCode::SysErrSenderStateInvalid,
                format!(
                    "cannot send value: {:?} exceeds balance: {:?}",
                    value, self.balance
                ),
            ));
        }

```
Makes more logical sense to me, thoughts? Removes the clone which doesn't really matter in a test but more readable imo